File: network/src/main/java/home/oleg/placesnearme/network/di/NetworkModule.java
package home.oleg.placesnearme.network.di;

import dagger.Module;
import dagger.Provides;
import home.oleg.placesnearme.network.config.NetworkConfig;
import home.oleg.placesnearme.network.service.FourSquareAPIFactory;
import home.oleg.placesnearme.network.service.IFourSquareAPI;
import io.reactivex.annotations.NonNull;

/**
 * Created by Oleg Sheliakin on 25.09.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
@Module
public final class NetworkModule {

    @Provides
    @NonNull
    public static IFourSquareAPI provideFourSquareApi(NetworkConfig config) {
        return FourSquareAPIFactory.create(config);
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/di/NetworkModule.kt
package home.oleg.placesnearme.coredata.di

import dagger.Module
import dagger.Provides
import home.oleg.placesnearme.corenetwork.config.NetworkConfig
import home.oleg.placesnearme.corenetwork.service.FourSquareAPIFactory
import home.oleg.placesnearme.corenetwork.service.IFourSquareAPI

@Module
object NetworkModule {

    @JvmStatic
    @Provides
    internal fun provideApi(networkConfig: NetworkConfig): IFourSquareAPI {
        return FourSquareAPIFactory.create(networkConfig)
    }

}
